main.c

```diff
@@ -1511,44 +1568,38 @@ static const struct xdg_toplevel_listener toplvl_listener = {
 static void configure(void *d, struct xdg_surface *surf, uint32_t serial)
 {
 	xdg_surface_ack_configure(surf, serial);
-	int col = term.confwidth / term.cwidth;
-	int row = term.confheight / term.cheight;
+	int s  = term.scale, m = (s + 119) / 120;
+	int cw = term.cwidth  * 120;
+	int ch = term.cheight * 120;
+	int col = term.confwidth/m * s/cw;
+	int row = term.confheight/m * s/ch;
 	struct winsize ws = {
 		row, col, 0, 0
 	};
 
 	assert(!term.configured);
 	term.configured = true;
-
 	if (col == 0 || row == 0)
 		return;
 
 	if (term.width == term.confwidth && term.height == term.confheight)
 		return;
-
+	term.width = term.confwidth;
+	term.height = term.confheight;
 	if (term.cfg.margin) {
-		term.width = term.confwidth;
-		term.height = term.confheight;
-		term.margin.left = (term.width - col * term.cwidth) / 2;
-		term.margin.top = (term.height - row * term.cheight) / 2;
-		term.need_redraw = true;
-		term.resize = 2;
-	} else {
-		term.width = col * term.cwidth;
-		term.height = row * term.cheight;
+		term.margin.left = (term.confwidth/m  * s - col * cw) / (2*120);
+		term.margin.top  = (term.confheight/m * s - row * ch) / (2*120);
```

Comment:
Why scale back and forth here, isn't it all in buffer coordinates?

Reply:
You can just use the actual size of the terminal here so it's all in buffer coordinates, you don't need confwidth. Without scaling they were equal so it didn't matter before.